### PR TITLE
Add SEED-Story

### PR DESCRIPTION
**Pull request by AndysonYs**
This PR adds https://github.com/TencentARC/SEED-Story, a multimodal story generation model, as a library.

### CODE REVIEW

File: packages/tasks/src/model-libraries.ts [hunk 1]
```diff
@@ -575,6 +575,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
...
 		countDownloads: `path_filename:"model" AND path_extension:"mil" AND _exists_:"path_prefix"`,
 	},
+	"seed-story": {
+		prettyLabel: "SEED-Story",
+		repoName: "SEED-Story",
+		repoUrl: "https://github.com/TencentARC/SEED-Story",
+		filter: false,
+		countDownloads: `path:"pytorch_model.bin" OR path:"tokenizer.model"`,
+		snippets: snippets.seed_story,
+	},
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts:585]: Can you move this section up to keep the alphabetical order in the file? Thanks!

File: packages/tasks/src/model-libraries.ts [hunk 2]
```diff
@@ -575,6 +575,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		snippets: snippets.whisperkit,
 		countDownloads: `path_filename:"model" AND path_extension:"mil" AND _exists_:"path_prefix"`,
 	},
+	"seed-story": {
+		prettyLabel: "SEED-Story",
+		repoName: "SEED-Story",
+		repoUrl: "https://github.com/TencentARC/SEED-Story",
+		filter: false,
+		countDownloads: `path:"pytorch_model.bin" OR path:"tokenizer.model"`,
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts:583]: Which files from https://huggingface.co/TencentARC/SEED-Story/tree/main are always downloaded when using this model? For example, is [cvlm_llama2_tokenizer/tokenizer.model](https://huggingface.co/TencentARC/SEED-Story/blob/main/cvlm_llama2_tokenizer/tokenizer.model) always downloaded? I'm asking because it's usually best to explicitly define a single file. The problem with defining `pytorch_model.bin` is that there are 4 (?) of them in the repo. If a user downloads the whole repo, it will be counted as 4 downloads which is not accurate.

**Review by Wauplin — COMMENTED** [review]
Thanks for opening this PR @AndysonYs! I've added a few comments to improve the integration. Let me know if you have any questions!

File: packages/tasks/src/model-libraries-snippets.ts [hunk 3]
```diff
@@ -813,4 +813,11 @@ whisperkit-cli transcribe --audio-path /path/to/audio.mp3
 # Or use your preferred model variant
 whisperkit-cli transcribe --model "large-v3" --model-prefix "distil" --audio-path /path/to/audio.mp3 --verbose`,
 ];
+
+export const seed_story = (): string[] => [
+	`# Install from https://github.com/TencentARC/SEED-Story.git
+
+# refer to https://github.com/TencentARC/SEED-Story/blob/master/configs/clm_models/agent_7b_sft.yaml on how to define args.
+# Use in SEED-Story (https://huggingface.co/TencentARC/SEED-Story) `,
```
[the remark this answers is not in the document]
Comment (AndysonYs) [reply packages/tasks/src/model-libraries-snippets.ts:821]: Thanks. I will provide an example.

File: packages/tasks/src/model-libraries.ts [hunk 4]
```diff
@@ -575,6 +575,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		snippets: snippets.whisperkit,
 		countDownloads: `path_filename:"model" AND path_extension:"mil" AND _exists_:"path_prefix"`,
 	},
+	"seed-story": {
```
[the remark this answers is not in the document]
Comment (AndysonYs) [reply packages/tasks/src/model-libraries.ts:578]: I have added these. BTW, my model pipeline can be defined as t2i, i2t, i2i, and t2t. Basically it is a multimodal generation model. So is there a proper pipeline for this?

File: packages/tasks/src/model-libraries.ts (hunk 2 above)
Comment (AndysonYs) [reply packages/tasks/src/model-libraries.ts:583]: [cvlm_llama2_tokenizer/tokenizer.model](https://huggingface.co/TencentARC/SEED-Story/blob/main/cvlm_llama2_tokenizer/tokenizer.model) is always downloaded. 

The seedx(https://huggingface.co/TencentARC/SEED-Story/tree/main/seedx) and detokenizer-pretrained(https://huggingface.co/TencentARC/SEED-Story/tree/main/detokenizer/detokenizer_pretrained) model is downloaded only when you want to train a new model. 

The seed-story(https://huggingface.co/TencentARC/SEED-Story/tree/main/seed_story) and detokenzier-adapted(https://huggingface.co/TencentARC/SEED-Story/tree/main/detokenizer/detokenizer_george_adapted)  is downloaded when you want to inference.

Do you have any suggestions in that case?

File: packages/tasks/src/model-libraries.ts (hunk 1 above)
Comment (AndysonYs) [reply packages/tasks/src/model-libraries.ts]: Sure. I've moved it.

File: packages/tasks/src/model-libraries.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (julien-c) [reply packages/tasks/src/model-libraries.ts]: maybe `text-to-image-text`?

File: packages/tasks/src/model-libraries.ts (hunk 2 above)
Comment (Wauplin) [reply packages/tasks/src/model-libraries.ts]: Then I think the best os to count on `path:"cvlm_llama2_tokenizer/tokenizer.model"` since it'll always be downloaded when user tries to run the model.

File: packages/tasks/src/model-libraries-snippets.ts [hunk 5]
```diff
@@ -401,6 +401,23 @@ export const timm = (model: ModelData): string[] => [
 model = timm.create_model("hf_hub:${model.id}", pretrained=True)`,
 ];
 
+export const seed_story = (): string[] => [
+	`# Install from https://github.com/TencentARC/SEED-Story.git
+# Use in SEED-Story (https://huggingface.co/TencentARC/SEED-Story)
```
Comment (osanseviero) [inline packages/tasks/src/model-libraries-snippets.ts:406]: Can be removed imo
1. There will be a link to the GitHub repo too
2. The snippet will only show in the SEED-Story repo

```suggestion
```

File: packages/tasks/src/model-libraries-snippets.ts [hunk 6]
```diff
@@ -401,6 +401,23 @@ export const timm = (model: ModelData): string[] => [
...
+# seed_story_cfg_path refers to 'https://github.com/TencentARC/SEED-Story/blob/master/configs/clm_models/agent_7b_sft.yaml'
+# llm_cfg_path refers to 'https://github.com/TencentARC/SEED-Story/blob/master/configs/clm_models/llama2chat7b_lora.yaml'
```
Comment (osanseviero) [inline packages/tasks/src/model-libraries-snippets.ts:408]: Fine for me to merge with this, but I feel these could be hosted under the HF repository, and then `OmegaConf.load("TencentARC/SEED-Story")` could load these configs directly from there

File: packages/tasks/src/model-libraries-snippets.ts [hunk 7]
```diff
@@ -401,6 +401,23 @@ export const timm = (model: ModelData): string[] => [
...
+from omegaconf import OmegaConf
+import hydra
+
+# load Llama2
+llm_cfg = OmegaConf.load(llm_cfg_path)
+llm = hydra.utils.instantiate(llm_cfg, torch_dtype=dtype_str)
```
Comment (osanseviero) [inline packages/tasks/src/model-libraries-snippets.ts:414]: Let's specify a dtype directly

```suggestion
llm = hydra.utils.instantiate(llm_cfg, torch_dtype="fp16")
```

File: packages/tasks/src/model-libraries.ts [hunk 8]
```diff
@@ -489,6 +489,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		filter: false,
 		countDownloads: `path_extension:"safetensors"`,
 	},
+	"seed-story": {
+		prettyLabel: "SEED-Story",
+		repoName: "SEED-Story",
+		repoUrl: "https://github.com/TencentARC/SEED-Story",
+		filter: false,
+		countDownloads: `path:"pytorch_model.bin" OR path:"tokenizer.model"`,
```
Comment (osanseviero) [inline packages/tasks/src/model-libraries.ts:497]: Maybe this based on the open discussion

```suggestion
		countDownloads: `path:"cvlm_llama2_tokenizer/tokenizer.model"`,
```

**Review by osanseviero — APPROVED** [review]
Sorry for the slow review! There are a couple small nits but seems close to be merge-able!

**Comment by AndysonYs** [comment]
@osanseviero Thanks. I have taken your suggestions and commit them. Is there anything i should do to complete this PR?

File: packages/tasks/src/model-libraries-snippets.ts [hunk 9]
```diff
@@ -401,6 +401,21 @@ export const timm = (model: ModelData): string[] => [
 model = timm.create_model("hf_hub:${model.id}", pretrained=True)`,
 ];
 
+export const seed_story = (): string[] => [
+# seed_story_cfg_path refers to 'https://github.com/TencentARC/SEED-Story/blob/master/configs/clm_models/agent_7b_sft.yaml'
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts:405]: there is a "`" missing I think?

File: packages/tasks/src/model-libraries-snippets.ts [hunk 10]
```diff
@@ -813,4 +828,5 @@ whisperkit-cli transcribe --audio-path /path/to/audio.mp3
 # Or use your preferred model variant
 whisperkit-cli transcribe --model "large-v3" --model-prefix "distil" --audio-path /path/to/audio.mp3 --verbose`,
 ];
+
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts:831]:
```suggestion
```

**Review by Wauplin — APPROVED** [review]
Thanks for the changes @AndysonYs. Looks almost good to me :)

(+ there is a merge conflict to resolve)

**Comment by AndysonYs** [comment]
@Wauplin Thanks. I have fixed all the above typos and the conflicts.

**Review by Wauplin — APPROVED** [review]
Perfect! Thanks @AndysonYs :hugs:
